src/assets/AssetsDetectionController.test.ts

```diff
@@ -429,6 +429,40 @@ describe('AssetsDetectionController', () => {
     ]);
   });
 
+  it('should update the tokens list when new tokens are detected', async () => {
```

Comment:
It seems like this test would pass before the fix as well :thinking: It seems useful - now we know that if `getBalancesInSingleCall` returns a distinct set of balances, the old and new balances are preserved. But I don't think it demonstrates the change being made in this PR.

The bug fix here affects the `tokensToDetect` array, which is only used as a parameter to `getBalancesInSingleCall`. So to test that the correct `tokensToDetect` are passed to `getBalancesInSingleCall`, I think we'll need to spy on `getBalancesInSingleCall` and ensure it's being called with the correct list. Or something along those lines.

It seems `getBalancesInSingleCall` is already in-scope as a spy, so that might not be too challenging.